Trunk 3401

gnyamach opened the pull request:
Hi Team,

Tell and I made some changes in the OpenMRS ProviderServiceImpl.java to allow users to be assigned to Providers, please review and accept the changes.

https://github.com/openmrs/openmrs-core/pull/2297/files

Thanks

teleivo committed:
Rename ModuleFileParserTest to UnitTest

teleivo committed:
Add tests for ModuleFileParser

* add ModuleFileParserTest which is BaseContextSensitive since the
ModuleFileParser makes use of Context
* test by creating an xml document and saving it in a Jar with extension
".omod" so its recognized as openmrs module
* also test using the logic-0.2 module file which is under the test
resources
* remove javadocs from ModuleFileParserUnit test as they just clutter

teleivo committed:
Upgrade jacoco-maven-plugin

teleivo committed:
TRUNK-5377 Remove redundant logging on failed module start

inline private method ModuleFactory.getModuleFromFile
since it does not add any value. (It is private and only used once.)

It simply catches ModuleExceptions, adds logging that
replicates whats inside the Exception and rethrows the ModuleException.
The same information will be visible in the log of the ModuleException
which causes the exception to be logged twice.

teleivo committed:
TRUNK-5385 Rearrange ModuleFileParser fields

and remove unnecessary set to null

teleivo committed:
Merge pull request #2627 from teleivo/module-file-parser-test

Test ModuleFileParser

mseaton committed:
TRUNK-5375 - up-porting to master

teleivo committed:
Merge pull request #2634 from teleivo/TRUNK-5377

TRUNK-5377 Remove redundant logging on failed module start

sumangala028 committed:
Added some test cases for FormServiceTest Class

mogoodrich committed:
TRUNK-5375: New Core 2.1 Cohort module significantly slows down Cohort manipulation

IsurangaPerera committed:
add unit tests

teleivo committed:
TRUNK-5381 ModuleFileParser invalid config message is wrong

remove misleading word locale from message key and pass valid
configVersion's along with the wrong configVersion set in the module's
config.xml which was parsed

teleivo committed:
Merge pull request #2620 from IsurangaPerera/TRUNK-5087

TRUNK-5087 : Add tests to OpenmrsUtil

klambo94 committed:
TRUNK-5259: Changed install.header.caption from OpenMRS {0} Installat… (#2617)

* TRUNK-5259: Changed install.header.caption from OpenMRS {0} Installation Wizard -> OpenMRS Platform {0} Installation Wizard

* TRUNK-5259: Updated messages.properties to use 'Core' over 'Platform'

teleivo committed:
TRUNK-5385 Remove unused parameters from ModuleFileParser's private methods

configVersion and jarFile were often not used in private methods
thus now removed

teleivo committed:
TRUNK-5385 Extract parsing of stream to xml document

to reduce the method size and move the same level of abstraction into a
method, also a step to prepare the use of try-with-resource

teleivo committed:
TRUNK-5385 Extract creation of module from document

method is concerned with creating an instance of Module
with values from the config xml Document

teleivo committed:
TRUNK-5385 Use try-with-resource on config xml InputStream

move getting the inputstream of the config jarentry into a
try-with-resource block since the InputStream is autocloseable the
stream will be closed for us and the closing logic in the finally can be
removed

teleivo committed:
TRUNK-5385 Move creation of Module outside of config jar/stream logic

getting the jarfile, jarentry for config.xml and parsing the config
stream into an xml document is separate from creating the module from
the xml document thus move the creation below it

teleivo committed:
TRUNK-5385 Use try-with-resource on JarFile

since its Autoclosable we can remove our finally closing logic and let
the try-with-resource do it

teleivo committed:
TRUNK-5385 Extract getting the xml config document into helper

and rearrange private methods in call order for readability

teleivo committed:
TRUNK-5385 Extract constant for module config.xml filename

teleivo committed:
TRUNK-5385 Extract getting zip entry and stream of config.xml

teleivo committed:
TRUNK-5385 Rename configDoc to config

we know its a Document from its type

teleivo committed:
TRUNK-5385 Use try-with-resource on ByteArrayOutputStream

to remove finally close() logic

teleivo committed:
TRUNK-5385 Rename config variable

teleivo committed:
TRUNK-5385 Extract method getElementTrimmed

since its often done

teleivo committed:
TRUNK-5385 Extract validateConfigVersion method

to increase readability of method, we do not need to read the details of
it when just looking at the creation of an instance of Module

shifona1 committed:
TRUNK-5230: Updated savePatientProgram to set the End date of all recent states of each workflow and added two test cases.

teleivo committed:
TRUNK-5385 Extract ensureValidConfigVersion

teleivo committed:
TRUNK-5385 Extract ensureNonEmptyName

so createModule simply reads as getXXXElementsContent and module.setXXX
if ones concerned with details, dig deeper into helper methods

teleivo committed:
TRUNK-5385 Extract ensureNonEmptyId/Package

teleivo committed:
TRUNK-5385 Remove duplication in getting an element or fail

extract duplicate code into method

teleivo committed:
TRUNK-5385 Declare module when created

declare the variable when its used

teleivo committed:
TRUNK-5385 Extract duplicate code which splits TagContentByWhitespace

teleivo committed:
TRUNK-5385 Group getElement helpers together

teleivo committed:
TRUNK-5385 Extract newDocumentBuilder

to keep level of abstraction in parseConfigXmlStream

mogoodrich committed:
TRUNK-5386: Revert method signature of Cohort addMember(Integer) method

teleivo committed:
TRUNK-5385 Add test parsing invalid module config.xml

teleivo committed:
TRUNK-5385 fix formatting in ModuleFileParserTest

teleivo committed:
TRUNK-5385 Test ModuleFileParser(InputStream)

teleivo committed:
TRUNK-5385 Use try-with-resource in ModuleFileParser(InputStream)

to auto close the OutputStream where the InputStream is written to
and extract private methods to improve readability

teleivo committed:
TRUNK-5385 Use junit ExpectedException in ModuleFileParserUnitTest

expecting on the substring of the exception messages links the test more
to its particular case. Since all 3 tests expect the same Exception it
would not surface if they test the same case, which already happened in
another class.

teleivo committed:
TRUNK-5385 Remove nesting when reading globalProperties

teleivo committed:
TRUNK-5385 Extract extractGlobalProperty

for readability and further refactoring

teleivo committed:
TRUNK-5385 Rename getGlobalProperties

to extractGlobalProperties to separate it from a normal getter

teleivo committed:
TRUNK-5385 Test that globalProperty text nodes are ignored

teleivo committed:
TRUNK-5385 Use Element when extracting globalProperty

getElementsByTagName returns type Node but they are all Element's
since we look for <globalProperty>, thus cast to Element

this enables us to only get child Element's we need without the need to
loop through all Node's

teleivo committed:
TRUNK-5385 Debug log globalProperty using format strings

no need to do string concatenation

teleivo committed:
TRUNK-5385 Extract removeTabsAndTrim

so we can simply wrap the string returned from getElementTrimmed and
clean it from tabs. The returned string from getElementTrimmed is never
null so no need to check. also added test to assert we can handle it.

teleivo committed:
TRUNK-5385 Extract method for creating GlobalProperty

teleivo committed:
TRUNK-5385 Make guard clause for empty property

this makes it easier to read that we exit if <globalProperty> doesnt
have a <property> child with a value

teleivo committed:
TRUNK-5385 Extract createGlobalPropertyWithDatatype

to hide details of how the GP is created when datatypeClassname is not
empty

teleivo committed:
TRUNK-5385 Rely on Element in getModuleToVersionMap

when using getElementsByTagName we know we get Element's and can thus
cast and simplify code because we know we only get 'modules' or 'module'
Element's

reduce nesting with a guard clause at the beginning of the method

teleivo committed:
TRUNK-5385 Refactor extracting advice

using the same pattern
* reduce nesting with a guard clause
* rely on getElementsByTag which returns actual Element
we are looking for
* use helper getElementTrimmed to hide getting element, defaulting to
empty string if not present, trimming it

teleivo committed:
TRUNK-5385 Harmonize method names extracting from xml

to be called extractGlobalProperties, extractConditionalResources
and group them together in createModule

teleivo committed:
TRUNK-5385 Refactor extracting extension and privilege

using the same pattern
* reduce nesting with a guard clause
* rely on getElementsByTag which returns actual Element
we are looking for
* use helper getElementTrimmed to hide getting element, defaulting to
empty string if not present, trimming it

teleivo committed:
TRUNK-5385 Extract constants for module file extension

teleivo committed:
TRUNK-5385 Debug log global property with format string

teleivo committed:
TRUNK-5385 Rename rootNode to configRoot

to signal that the root is the root element of the module config which
might be obvious but its still easier to read and no need to always make
the mental connection

teleivo committed:
TRUNK-5385 Move extractConditionalResources in call order

since its called after extractMandatory move it below it

teleivo committed:
TRUNK-5385 Harmonize extractConditionalResource tests

move tests for extractConditionalResource into ModuleFileParserTest to
use same style of tests against the public methods instead of package
private one to prepare making it private

after refactorings where dependencies of ModuleFileParser will be put
into its constructor ModuleFileParser(MessageSourceService), the tests
can be written as BaseContextMockTests without use of a database

teleivo committed:
TRUNK-5385 Harmonize method parameter order

put Element first in extractModulesWithVersionAttribute as all other
methods which take more than one parameter with on of them of type Element
have it first

teleivo committed:
TRUNK-5385 Fix ModuleFileParser test parsing GP without description

it does not get ignored

teleivo committed:
Merge pull request #2644 from teleivo/TRUNK-5385

TRUNK-5385 Use try-with-resource and remove duplication in ModuleFileParser

teleivo committed:
Show ModuleFileParser allows empty conditionalResource elements

the ModuleFileParser currently allows

<conditionalResource>
</conditionalResource>

which leads to a ModuleConditionalResource which has no path and no
version. This also shows that the ModuleConditionalResource is flawed
since it only has a default constructor which leads to a state that
should not be allowed

shiddy reviewed (approved):
LGTM. Meets specifications. 👍

teleivo committed:
Test AdministrationService.addGlobalPropertyListener

by adding a unit test to the AdministrationServiceUnitTest
mocking EventListeners since
AdministrationServiceImpl.addGlobalPropertyListener
simply delegates to the EventListeners which serves its
List<GlobalPropertyListeners> to the public

alicerowan committed:
Renamed concept_numeric DB field "precise" to "allow_decimal"

PKatGITHUB committed:
TRUNK-5065: Add tests to AdministrationServiceImpl

mogoodrich committed:
Merge pull request #2645 from GME-Project/TRUNK-5333

TRUNK-5333 Rename concept_numeric DB field "precise" to "allow_decimal"

teleivo committed:
Merge pull request #2648 from PKatGITHUB/TRUNK-5065

TRUNK-5065: Add tests to AdministrationServiceImpl

teleivo committed:
Use static imports in AdministrationServiceTest

to improve readability since we use these asserts over and over again
its clearer to read `assertTrue` than `Assert.assertTrue`

teleivo committed:
Remove boilerplate javadocs in AdministrationServiceTest

these javadocs are often out of date, just take up space and repeat the
method signature that is under test anyway so navigating to it is easy
already via the test itself

teleivo committed:
Set AdminService before each test

let the @Before set the adminService which is under test each time,
tests should be independent

teleivo committed:
Use adminService from test setup in AdminServiceTest

we prepare the adminService in the @Before method for every test so
there is no need to get it from the Context again in each test

teleivo committed:
TRUNK-5065 Test saving allowed locales without default

add test that shows its not allowed to save a locale list via global
property if the list doesnt contain the default locale currently set.

paradisekelechi committed:
TRUNK-5338: make verification_status field nullable for the condition table (#2651)

paradisekelechi committed:
TRUNK-5360: Add changeset to migrate data to the new condition table (#2637)

wluyima committed:
Merge pull request #2635 from sumangala028/TRUNK-5061

TRUNK-5061 Added some test cases for FormServiceTest Class

mogoodrich committed:
Merge pull request #2643 from shifona1/TRUNK-5230

Trunk-5230: PatientProgram: When completing a program, the end date of the most recent state in each workflow should be set to the completion date of the program

mogoodrich committed:
TRUNK-5228: PatientProgram: If a program is completed, voidPatientState should set end date of previous state to date completed of program
TRUNK-5229: PatientProgram: If you transition to a new state when a program is complete, the state end date should be set to the completion date of the program

PKatGITHUB committed:
TRUNK-5065: Add tests to AdministrationServiceImpl

teleivo committed:
Merge pull request #2650 from PKatGITHUB/TRUNK-5065

TRUNK-5065: Add tests to AdministrationServiceImpl

gnyamach committed:
TRUNK-3401 Changed the messages thrown by the exceptions, the follow of the method, added tests, and made some adjustments recommended by the members of OpenMRS
